tests/chainer_tests/functions_tests/math_tests/test_erf.py

```diff
@@ -0,0 +1,107 @@
+import math
+import unittest
+
+import numpy
+
+import chainer
+from chainer import cuda
+import chainer.functions as F
+from chainer import gradient_check
+from chainer import testing
+from chainer.testing import attr
+
+
+def _erf_cpu(x):
+    return numpy.vectorize(math.erf)(x)
+
+
+def _erf_gpu(x):
+    return cuda.to_gpu(_erf_cpu(cuda.to_cpu(x)))
+
+
+class UnaryFunctionsTestBase(unittest.TestCase):
+
+    def make_data(self):
+        raise NotImplementedError
+
+    def setUp(self):
+        self.x, self.gy = self.make_data()
+        if self.dtype == numpy.float16:
+            self.check_forward_options = {'atol': 1e-3, 'rtol': 1e-3}
+            self.check_backward_options = {'atol': 3e-3, 'rtol': 1e-2}
+            self.check_double_backward_options = {'atol': 3e-3, 'rtol': 1e-2}
+        else:
+            self.check_forward_options = {'atol': 1e-7, 'rtol': 1e-7}
+            self.check_backward_options = {'atol': 1e-4, 'rtol': 1e-3}
+            self.check_double_backward_options = {'atol': 1e-4, 'rtol': 1e-3}
+
+    def check_forward(self, op, op_np, x_data):
+        x = chainer.Variable(x_data)
+        y = op(x)
+        self.assertEqual(x.data.dtype, y.data.dtype)
+        testing.assert_allclose(
+            op_np(self.x), y.data, **self.check_forward_options)
+
+    def check_forward_cpu(self, op, op_np):
+        self.check_forward(op, op_np, self.x)
+
+    def check_forward_gpu(self, op, op_np):
+        self.check_forward(op, op_np, cuda.to_gpu(self.x))
+
+    def check_backward(self, op, x_data, y_grad):
+        gradient_check.check_backward(
+            op, x_data, y_grad, dtype=numpy.float64,
+            **self.check_backward_options)
+
+    def check_backward_cpu(self, op):
+        self.check_backward(op, self.x, self.gy)
+
+    def check_backward_gpu(self, op):
+        self.check_backward(op, cuda.to_gpu(self.x), cuda.to_gpu(self.gy))
+
+    def check_double_backward(self, op, x_data, y_grad, y_grad_grad):
+        gradient_check.check_double_backward(
+            op, x_data, y_grad, y_grad_grad, dtype=numpy.float64,
+            **self.check_double_backward_options)
+
+    def check_double_backward_cpu(self, op):
+        self.check_double_backward(op, self.x, self.gy, self.ggy)
+
+    def check_double_backward_gpu(self, op):
+        self.check_double_backward(op, cuda.to_gpu(
+            self.x), cuda.to_gpu(self.gy), cuda.to_gpu(self.ggy))
+
+    def check_label(self, op, expected):
+        self.assertEqual(op().label, expected)
+
+
+@testing.parameterize(*testing.product({
+    'shape': [(3, 2), ()],
+    'dtype': [numpy.float16, numpy.float32, numpy.float64],
+}))
+class TestErf(UnaryFunctionsTestBase):
```

Comment:
Could you consider using `unary_math_function_unittest` utility to generate test cases? See the usage here: https://docs.chainer.org/en/stable/reference/generated/chainer.testing.unary_math_function_unittest.html#chainer.testing.unary_math_function_unittest